### DISCUSSION TITLE
Support other postgres clients compatible with drizzle

### DISCUSSION THREAD

**Discussion by tompeace** [payloadcms/payload#5874] [Feature Requests & Ideas]
Firstly I think the move to drizzle was a really smart one. The ability drizzle has to use the best client for the users needs is a very powerful one and one that seems very in line with PayloadCMS stated goals surrounding DX. 

However, at the moment we only have the ability to use node-postgres as a postgres client. I would like to be able to use Payload with the [RDS Data client](https://orm.drizzle.team/docs/get-started-postgresql#aws-data-api) for enterprise/security reasons and so would love to discuss be able to customise this!

The main problem I see is that `drizzle` itself is an adapter. Therefore `connect.ts` would probably need to receive the drizzle object itself instead of creating it w/ `new Pool()`, or at least skipping this process if a drizzleClient is passed directly.

This could probably look something like this 
```
import drizzle from 'drizzle-orm/aws-data-api/pg'
const drizzleClient = drizzle(new RDSDataClient({}), {})
...
export default buildConfig({
  db: postgresAdapter({ drizzleClient }),
})
```

I have been using drizzle on quite a few projects, with D1 (I know you dont support sqlite yet), and Neon and have to say the experience has been pretty smooth. 
Would love to hear your thoughts!

**Comment by jmikrut** [comment]
100%. This is necessary and we want to do it as well.

Got some plans for how to integrate actually. We are going to be abstracting the core relational DB logic out of the `db-postgres` package so we can re-use it for SQLite, and then from there, supporting other Drizzle Postgres implementations will follow the same pattern.

Expect more from us here in the near future. I am hoping to begin this work in the next few weeks actually.

**Reply by tompeace** [reply]
Awesome, @jmikrut am happy to hear that!

**Reply by dansmachina** [reply]
@jmikrut do you have any updates about that? I'd would be really a game changer, we are also looking into using, for example AWS Data API Postgres but so far I couldn't find an smart way to do it out of the box. 
Thanks a lot!
